### PR TITLE
Fix packaging

### LINKED ISSUE #88: Setuptools warnings detected for newer python versions, python-3.8, 3.9
build log snipits:

writing manifest file 'Tubes.egg-info/SOURCES.txt'
/usr/lib/python3.8/site-packages/setuptools/dist.py:153: UserWarning: newlines not allowed and will break in the future
  warnings.warn("newlines not allowed and will break in the future")
...
running install_scripts
/usr/lib/python3.8/site-packages/setuptools/dist.py:153: UserWarning: newlines not allowed and will break in the future
  warnings.warn("newlines not allowed and will break in the future")
...
writing Tubes.egg-info/PKG-INFO
/usr/lib/python3.9/site-packages/setuptools/dist.py:153: UserWarning: newlines not allowed and will break in the future
  warnings.warn("newlines not allowed and will break in the future")
writing dependency_links to Tubes.egg-info/dependency_links.txt
writing requirements to Tubes.egg-info/requires.txt
writing top-level names to Tubes.egg-info/top_level.txt
...
running install_egg_info
running egg_info
writing Tubes.egg-info/PKG-INFO
/usr/lib/python3.9/site-packages/setuptools/dist.py:153: UserWarning: newlines not allowed and will break in the future
  warnings.warn("newlines not allowed and will break in the future")

##########

Full build logs available at Gentoo's Bugzilla

https://bugs.gentoo.org/797271
https://bugs.gentoo.org/attachment.cgi?id=717297

### PR DESCRIPTION
**Pull request by graingert**
Fixes #88

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -1,5 +1,24 @@
+[metadata]
+name = Tubes
+version = 0.2.0
+description = Flow control and backpressure for event-driven applications.
+license = MIT
+url = https://github.com/twisted/tubes/
+
+[options]
+packages = find:
+package_dir = tubes=tubes
```
Comment (graingert) [inline setup.cfg]:
```suggestion
```

I don't think this is needed with `package = find:`

File: setup.cfg [hunk 2]
```diff
@@ -1,5 +1,24 @@
...
+install_requires =
+    six
+    Twisted
+include_package_data = True
+
+[options.packages.find]
+exclude =
```
Comment (graingert) [inline setup.cfg]: I'm pretty sure an empty exclude is the default
```suggestion
```

File: setup.cfg (hunk 1 above)
Comment (graingert) [reply setup.cfg]: Indeed this config is no different to the default and so I removed it:
https://docs.python.org/3/distutils/setupscript.html#listing-whole-packages

>  The default correspondence is the most obvious one, i.e. package distutils is found in the directory distutils relative to the distribution root.

File: setup.cfg (hunk 2 above)
Comment (graingert) [reply setup.cfg]: indeed, so removed

9:15 AM @graingert: and `[options.packages.find] exclude` defaults to empty?
9:17 AM @uranusjr: yes

File: setup.py [hunk 3]
```diff
@@ -1,23 +1,2 @@
...
-
-    packages=find_packages(exclude=[]),
-    package_dir={'tubes': 'tubes'},
-    install_requires=[
-        "six",
-        "Twisted",
-    ],
-    include_package_data=True,
-)
+from setuptools import setup
```
Comment (webknjaz) [inline setup.py:1]: Note that starting with pip 21.1 you don't even need this boilerplate: https://github.com/pypa/setuptools/pull/2871.

File: setup.cfg [hunk 4]
```diff
@@ -1,5 +1,18 @@
+[metadata]
+name = Tubes
+version = 0.2.0
+description = Flow control and backpressure for event-driven applications.
+license = MIT
+url = https://github.com/twisted/tubes/
+
+[options]
```
Comment (webknjaz) [inline setup.cfg:8]: While you're on it, maybe add `python_requires` too? Could be in a follow-up PR tho.

File: setup.cfg [hunk 5]
```diff
@@ -1,5 +1,18 @@
...
+packages = find:
+install_requires =
+    six
+    Twisted
+
 [bdist_wheel]
-universal=1
+universal = 1
```
Comment (webknjaz) [inline setup.cfg:15]: FTR this produces a `py2.py3` tag. Does this project still support Python 2? If not, this should be `0` (or removed).

File: setup.py [hunk 6]
```diff
@@ -1,23 +1,2 @@
...
-    license="MIT",
-    url="https://github.com/twisted/tubes/",
-
-    packages=find_packages(exclude=[]),
-    package_dir={'tubes': 'tubes'},
-    install_requires=[
-        "six",
-        "Twisted",
-    ],
-    include_package_data=True,
```
Comment (webknjaz) [inline setup.py:22]: This seems to be lost. OTOH, it may be unnecessary if there's no package data files to ship.

File: setup.py (hunk 3 above)
Comment (webknjaz) [reply setup.py:1]: Plus it's best to explicitly declare the build backend in `pyproject.toml`.

File: setup.cfg [hunk 7]
```diff
@@ -1,5 +1,18 @@
...
 
 [flake8]
-ignore=E302,E303,E402
+ignore = E302,E303,E402
```
Comment (webknjaz) [inline setup.cfg:18]: Should this be `extend-ignore`?

File: setup.cfg (hunk 4 above)
Comment (graingert) [reply setup.cfg:8]: I think tubes still supports python 2 (hence the six dep)

File: setup.cfg (hunk 7 above)
Comment (graingert) [reply setup.cfg:18]: Ah yep this should be extend-ignore

File: setup.py (hunk 6 above)
Comment (graingert) [reply setup.py:22]: I removed it because there's no package data

File: setup.py (hunk 3 above)
Comment (glyph) [reply setup.py:1]: We can have a separate issue for that, I think.

File: setup.cfg (hunk 7 above)
Comment (glyph) [reply setup.cfg:18]: @graingert can you update the PR?

File: setup.cfg (hunk 5 above)
Comment (glyph) [reply setup.cfg:15]: It still does for now but it should probably be removed soon after landing this.

**Comment by glyph** [comment]
@graingert hey we've actually got CI now

File: setup.cfg (hunk 7 above)
Comment (graingert) [reply setup.cfg:18]: I think we should fix this in a follow up. My plan was to make no semantic changes

**Review by glyph — COMMENTED** [review]
👍

**Comment by dol-sen** [comment]
@glyph are you making a release in the near future?  I've seen you have been making some commits in the past few days.

**Comment by glyph** [comment]
> @glyph are you making a release in the near future?  I've seen you have been making some commits in the past few days.

Possibly. Why do you ask?

**Comment by dol-sen** [comment]
I'll wait to make a pkg bump for gentoo with current HEAD otherwise.
If your going to make a release soon, I'll wait for the release.  It is a dependency for klein, needed for buildbot.
I was just doing a round of pkg maintenance.

**Comment by glyph** [comment]
> I'll wait to make a pkg bump for gentoo with current HEAD otherwise.

Does a 'pkg bump' mean *shipping* current HEAD?

> If your going to make a release soon, I'll wait for the release. It is a dependency for klein, needed for buildbot.
> I was just doing a round of pkg maintenance.

Since Tubes's process is pretty lightweight, I just made a v0.2.1 release to avoid any shipping of random untested git hashes.

**Comment by dol-sen** [comment]
Awesome, thank you

And yes, a bump otherwise would have been fixed to a current commit hash.
Which would have been better than carrying patches to apply to the previous
release.

**Comment by glyph** [comment]
Thanks for reaching out; why were you carrying patches?

**Comment by dol-sen** [comment]
I only had the one patch for the 0.2.0 release, the unnecessary dep patch.

Typically we try to not carry patches, but some are needed for some pkgs to
make them compatible with our build environment, like disabling some tests
that require internet (which we generally are in a sandboxed environment).

Looking at current twisted ebuild, there are 2 patches.   Yes, we usually
try to upstream any patches.

PATCHES=(
# https://twistedmatrix.com/trac/ticket/10200
"${FILESDIR}/${PN}-22.1.0-force-gtk3.patch"
# https://github.com/twisted/twisted/pull/1723
"${FILESDIR}/${P}-py311.patch"
)

**Pull request merged**